spec/client_spec.js

```diff
@@ -435,19 +480,48 @@ describe('Client', function() {
     });
 
     describe('#invoke', function() {
-      it('returns a promise', function() {
-        promise = subject.invoke('iframe.resize');
+      it('returns a promise', function(done) {
+        // in reality appendText doesn't return anything
+        promise = subject.invoke('ticket.appendText', 'foobar');
 
         expect(promise).to.be.a.promise;
+        expect(promise).to.eventually.become({ errors: {}, 'ticket.appendText': true }).and.notify(done);
+
+        window.addEventListener.callArgWith(1, {
+          origin: subject._origin,
+          source: subject._source,
+          data: { id: requestsCount, result: { errors: {}, 'ticket.appendText': true } }
+        });
       });
 
-      it('throw when called with an object', function() {
+      it('throws an error when the handler throws it', function(done) {
+        promise = subject.invoke('ticket.foo', 'bar');
+
+        expect(promise).to.be.a.promise;
+        expect(promise).to.be.rejectedWith(Error, 'ticket.foo unavailable').and.notify(done);
+
+        window.addEventListener.callArgWith(1, {
+          origin: subject._origin,
+          source: subject._source,
+          data: { id: requestsCount, result: { errors: { 'ticket.foo': { message: 'ticket.foo unavailable' } } } }
+        });
+      });
+
+      it('throws an error when invoked with an object', function() {
         expect(function() {
           subject.invoke({
             'iframe.resize': [1]
           });
         }).to.throw(Error);
```

Comment:
It has a good messge, it just that this test stubs it all. in reality the message would be the error message from console.